sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala

```diff
@@ -931,7 +941,8 @@ case class RocksDBCheckpointMetadata(
     sstFiles: Seq[RocksDBSstFile],
     logFiles: Seq[RocksDBLogFile],
     numKeys: Long,
-    columnFamilyMapping: Option[Map[String, Short]] = None,
+    numInternalKeys: Long,
+    columnFamilyMapping: Option[Map[String, ColumnFamilyInfo]] = None,
```

Comment:
This is a breaking change as the existing field gets different representation for json. Shall we check whether we ever release the version before the change, to ensure we are not breaking the checkpoint? Let's also check with the latest bugfix version of Spark 3.5.

Reply:
If we didn't have the test for compatibility then we probably should have it.

Reply:
I don't think we had this in 3.5 - do we still need a compatibility test then ?